Dockerfile

```diff
@@ -1,18 +1,53 @@
-FROM ubuntu:bionic
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
+# Build stage: Install ruby dependencies
+# ===
+FROM ruby:2.7 AS build-site
+WORKDIR /srv
+ADD . .
+RUN bundle install
+RUN bundle exec jekyll build
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:12-slim AS yarn-dependencies
+WORKDIR /srv
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
+
+# Build stage: Run "yarn run build-js"
+# ===
+FROM yarn-dependencies AS build-js
+ADD . .
+RUN yarn run build-js
+
+# Build stage: Run "yarn run build-css"
+# ===
+FROM yarn-dependencies AS build-css
+ADD . .
+RUN yarn run build-css
 
-# Set git commit ID
-ARG REVISION_ID
-RUN test -n "${REVISION_ID}"
+# Build the production image
+# ===
+FROM ubuntu:focal
 
-# Copy over files
+# Set up environment
+ENV LANG C.UTF-8
 WORKDIR /srv
-ADD _site .
+
+# Install nginx
+RUN apt-get update && apt-get install --no-install-recommends --yes nginx
+
+# Import code, build assets and mirror list
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js requirements.txt
```

Comment:
You don't need to rm the file `requirements.txt` since this is not a python project :)

You can remove `Gemfile.lock` instead.